crates/graphql_artifact_generation/src/iso_overload_file.rs

```diff
@@ -158,39 +168,66 @@ type MatchesWhitespaceAndString<
         content.push_str(&field_overload);
     }
 
-    let entrypoint_overloads = sorted_entrypoints(schema)
-        .into_iter()
-        .map(|field| build_iso_overload_for_entrypoint(field, file_extensions, no_babel_transform));
+    let entrypoint_overloads = sorted_entrypoints(schema).into_iter().map(|(field, _)| {
+        build_iso_overload_for_entrypoint(field, file_extensions, no_babel_transform)
+    });
     for (import, entrypoint_overload) in entrypoint_overloads {
         if let Some(import) = import {
             imports.push_str(&import);
         }
         content.push_str(&entrypoint_overload);
     }
 
-    content.push_str(
-        "
+    (match no_babel_transform {
+        false => {
+            content.push_str(
+                "
 export function iso(_isographLiteralText: string):
   | IdentityWithParam<any>
   | IdentityWithParamComponent<any>
   | IsographEntrypoint<any, any>
 {\n",
-    );
-
-    content.push_str(match no_babel_transform {
-        false => {
-"  throw new Error('iso: Unexpected invocation at runtime. Either the Babel transform ' +
+            );
+            content.push_str("  throw new Error('iso: Unexpected invocation at runtime. Either the Babel transform ' +
       'was not set up, or it failed to identify this call site. Make sure it ' +
       'is being used verbatim as `iso`. If you cannot use the babel transform, ' + 
-      'set options.no_babel_transform to true in your Isograph config. ');"
+      'set options.no_babel_transform to true in your Isograph config. ');\n}")
         }
         true => {
-            "  return (clientFieldResolver: any) => clientFieldResolver;"
+            let switch_cases =
+                sorted_entrypoints(schema)
+                    .into_iter()
+                    .map(|(field, iso_literal_text)| {
+                        format!(
+                            "    case '{}':
+      return import('./{}/{}/entrypoint{}').then(module => module.default);\n",
+                            iso_literal_text,
+                            field.type_and_field.type_name,
+                            field.type_and_field.field_name,
+                            file_extensions.ts()
+                        )
+                    });
+
+            content.push_str(
+                "
+export function iso(isographLiteralText: string):
+  | IdentityWithParam<any>
+  | IdentityWithParamComponent<any>
+  | Promise<IsographEntrypoint<any, any>>
+{
+  switch (isographLiteralText) {\n",
+            );
+
+            for switch_case in switch_cases {
```

Comment:
once we have directives, this won't quite be right...
In particular, we can have an `entrypoint Query.HomePage @lazy` or something in one file, and a `entrypoint Query.HomePage` in another.

Anyway, no big deal, we can cross that bridge later (since that will mean we probably can't have a single `entrypoint.ts` file anymore